[19.01] Only check for exit code on actual sort command

mvdbeek opened the pull request:
The grep command that collects headers will exit with a non-zero exit code if there is no header, so we can't use `subprocess.check_call` there. You can see the test is actually failing here: https://jenkins.galaxyproject.org/view/All/job/docker-main-tools/218/testReport/

And you can manually test the tool with planemo:
```
All 4 test(s) executed passed.
sort1[0]: passed
sort1[1]: passed
sort1[2]: passed
sort1[3]: passed
```

mvdbeek committed:
Only check for exit code on actual sort command

nsoranzo commented on tools/filters/sorter.py:
```diff
@@ -45,7 +45,7 @@ def main():
         # grep comments
         grep_comments = "(grep '^#' %s) >> %s" % (input, output)
-        subprocess.check_call(grep_comments, shell=True)
+        subprocess.call(grep_comments, shell=True)
```
You could capture the return value of `subprocess.call()` and call `stop_err()` if it is not 0 or 1.

Also, I'd suggest to:
- pass the 2 above commands as lists instead of strings, and remove `shell=True`
- replace the output redirects with the `stdout` parameter of `call`/`check_call`

mvdbeek committed:
Cleanup sorter.py

Feed open output file to Popen instead of using stdout redirection in
command string.

bernt-matthias reviewed (approved):
looks good to me

nsoranzo committed:
Restore check_call for `sed`
